lib/Dialect/TTIR/IR/TTIROps.cpp

```diff
@@ -596,6 +596,99 @@ ::mlir::LogicalResult mlir::tt::ttir::IndexOp::verify() {
 }
 // ANCHOR_END: decomposing_an_op_index_ttir_verify
 
+//===----------------------------------------------------------------------===//
+// SelectOp
+//===----------------------------------------------------------------------===//
+
+// SelectOp verification
+::mlir::LogicalResult mlir::tt::ttir::SelectOp::verify() {
+  ::mlir::RankedTensorType inputType = getInput().getType();
+  ::mlir::RankedTensorType outputType = getOutput().getType();
+
+  if (inputType.getRank() != outputType.getRank()) {
+    return emitOpError("Input and output tensors must have the same rank.");
+  }
+
+  if (inputType.getElementType() != outputType.getElementType()) {
+    return emitOpError("Input and output tensors must have the same element "
+                       "type.");
+  }
+
+  int32_t dim = getDim();
+  if (dim < 0) {
+    dim += inputType.getRank();
+  }
+
+  if (dim < 0 || dim >= inputType.getRank()) {
+    return emitOpError() << "Invalid dimension " << dim
+                         << " for select op with input tensor rank "
+                         << inputType.getRank();
+  }
```

Comment:
Can we wrap this into a utility method? Seems like we'd want this evaluation pattern for many ops.

Reply:
We can wrap it, but we want to return an error from this `verify` function. So this means that if you had 
```
auto dim = getRealDim(op, tensor, dim);
if (outOfBounds(dim, tensor)) 
  emitOpError() ...
```
you still have to have this kind of check here. 

Best way to utilize this I think is 

``` 
utils.cpp:
template <typename SizeT = int32_t>
std::enable_if_t<std::is_signed<SizeT>::value, llvm::LogicalResult>
getRealDim(mlir::Operation *op, mlir::RankedTensorType tensorType, SizeT &dim);

----
  
ops.cpp:
int32_t dim = getDim();
  if (failed(ttmlir::utils::getRealDim(*this, inputType, dim))) {
    return failure();
  }
```

do you think we should have it or not?

Reply:
You're right, I think I actually prefer the as-is version more than these two you outlined, let's leave it like this.